src/core/checkedint.d

```diff
@@ -168,7 +168,9 @@ pragma(inline, true)
 uint addu()(uint x, uint y, ref bool overflow)
 {
     immutable uint r = x + y;
-    if (r < x || r < y)
+    immutable bool o = r < x;
+    assert(o == (r < y));
+    if (o)
```

Comment:
Please rewrite to:

```d
if (r < x)
{
     assert(r < y);
     overflow = true;
}

Reply:
I had thought about that:

- Advantage: The new `o` object is not introduced.

- Dis-advantage: The reverse implication: `if (r >= x) assert(r >= y);` is not tested or (more importantly) self-documented via the `assert`.

I believe the dis-advantage significantly outweights the advantage.

Is there any other advantage, please ?

Reply:
You can do:

```d
if (r < x)
{
     assert(r < y);
     overflow = true;
}
assert(r >= y);
```